test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/IPv6ScopeIdPresentConditionAttribute.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Net.NetworkInformation;
+using System.Net.Sockets;
+using Microsoft.AspNetCore.Testing.xunit;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.FunctionalTests
+{
+    public class IPv6ScopeIdPresentConditionAttribute : Attribute, ITestCondition
+    {
+        private static readonly Lazy<bool> _ipv6Supported = new Lazy<bool>(CanBindToIPv6Address);
+
+        public bool IsMet
+        {
+            get
+            {
+                return _ipv6Supported.Value;
+            }
+        }
+
+        public string SkipReason
```

Comment:
Can be `public string SkipReason { get; } = "No IPv6 addresses with scope IDs were found on the host.";`